src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/AdvancedCargoOutputNode.java

```diff
@@ -42,6 +44,122 @@ public void init() {
 			@Override
 			public void newInstance(final BlockMenu menu, final Block b) {
 				try {
+
+					if (BlockStorage.hasBlockInfo(b)) {
+						if (Boolean.valueOf(BlockStorage.getBlockInfo(b, "furnace"))) {
+							String slot = BlockStorage.getBlockInfo(b, "furnace-slot");
+							menu.replaceExistingItem(7, new CustomItem(new MaterialData(Material.FURNACE), "&7Choose Slot", "", "&eCurrently: " + slot));
+							menu.addMenuClickHandler(7, new MenuClickHandler() {
+								@Override
+								public boolean onClick(Player player, int i, ItemStack itemStack, ClickAction clickAction) {
+									int[] border = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26};
```

Comment:
This variable should be final and static and defined at class-level.
Defining a new array everytime the menu is clicked, puts performance at risk.